projects/graph/src/draw.py

```diff
@@ -10,5 +12,60 @@
 
 class BokehGraph:
     """Class that takes a graph and exposes drawing methods."""
-    def __init__(self):
-        pass  # TODO
+    def __init__(self, graph):
+        self.graph = graph
+    def draw(self):
+        graph = self.graph
+        N = 8
+        node_indices = list(range(N))
+
+        plot = figure(title="Graph Layout Demonstration", x_range=(-1.1,1.1), y_range=(-1.1,1.1),
+                    tools="", toolbar_location=None)
+
+        graph = GraphRenderer()
+
+        graph.node_renderer.data_source.add(node_indices, 'index')
+        # graph_render.node_renderer.data_source.add(Spectral4, 'color')
+        graph.node_renderer.glyph = Circle(radius=0.25, fill_color='green')
+
+        graph.edge_renderer.data_source.data = dict(
+            start=[0]*N,
+            end=node_indices)
+
+        ### start of layout code
+        circ = [i*2*math.pi/8 for i in node_indices]
+        x = [math.cos(i) for i in circ]
+        y = [math.sin(i) for i in circ]
+        graph_layout = dict(zip(node_indices, zip(x, y)))
+        graph.layout_provider = StaticLayoutProvider(graph_layout=graph_layout)
```

Comment:
Good setup here; see what you can do to change the code just a bit to see how it changes your graph shape.